plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/delete/DeleteProjectAction.java

```diff
@@ -0,0 +1,215 @@
+import org.eclipse.che.ide.api.action.ActionEvent;
+import org.eclipse.che.ide.api.app.AppContext;
+import org.eclipse.che.ide.api.app.CurrentProject;
+import org.eclipse.che.ide.api.notification.NotificationManager;
+import org.eclipse.che.ide.dto.DtoFactory;
+import org.eclipse.che.ide.ext.openshift.client.OpenshiftLocalizationConstant;
+import org.eclipse.che.ide.ext.openshift.client.OpenshiftServiceClient;
+import org.eclipse.che.ide.ext.openshift.client.oauth.OpenshiftAuthorizationHandler;
+import org.eclipse.che.ide.ext.openshift.shared.dto.BuildConfig;
+import org.eclipse.che.ide.rest.DtoUnmarshallerFactory;
+import org.eclipse.che.ide.ui.dialogs.ConfirmCallback;
+import org.eclipse.che.ide.ui.dialogs.DialogFactory;
+
+import javax.validation.constraints.NotNull;
+import java.util.Collections;
+import java.util.List;
+
+import static org.eclipse.che.api.promises.client.callback.PromiseHelper.newCallback;
+import static org.eclipse.che.api.promises.client.callback.PromiseHelper.newPromise;
+import static org.eclipse.che.ide.ext.openshift.shared.OpenshiftProjectTypeConstants.OPENSHIFT_PROJECT_TYPE_ID;
+import static org.eclipse.che.ide.workspace.perspectives.project.ProjectPerspective.PROJECT_PERSPECTIVE_ID;
+import static org.eclipse.che.ide.ext.openshift.shared.OpenshiftProjectTypeConstants.OPENSHIFT_NAMESPACE_VARIABLE_NAME;
+import static org.eclipse.che.ide.ext.openshift.shared.OpenshiftProjectTypeConstants.OPENSHIFT_APPLICATION_VARIABLE_NAME;
+
+/**
+ * Action for deletion project from OpenShift
+ * @author Alexander Andrienko
+ */
+@Singleton
+public class DeleteProjectAction extends AbstractPerspectiveAction {
+
+    private final OpenshiftLocalizationConstant locale;
+    private final AppContext                    appContext;
+    private final OpenshiftAuthorizationHandler openshiftAuthorizationHandler;
+    private final DialogFactory                 dialogFactory;
+    private final OpenshiftServiceClient        service;
+    private final NotificationManager           notificationManager;
+    private final ProjectServiceClient          projectService;
+    private final DtoFactory                    dtoFactory;
+    private final DtoUnmarshallerFactory       dtoUnmarshaller;
+
+    @Inject
+    public DeleteProjectAction(OpenshiftLocalizationConstant locale,
+                               AppContext appContext,
+                               OpenshiftAuthorizationHandler openshiftAuthorizationHandler,
+                               DialogFactory dialogFactory,
+                               OpenshiftServiceClient service,
+                               ProjectServiceClient projectService,
+                               NotificationManager notificationManager,
+                               DtoFactory dtoFactory,
+                               DtoUnmarshallerFactory dtoUnmarshaller) {
+        super(Collections.singletonList(PROJECT_PERSPECTIVE_ID),
+              locale.deleteProjectTitle(),
```

Comment:
deleteProjectActionDescription but deleteProjectTitle not deleteProjectActionTitle. Why? It would be better to make they the same way